code/modules/cargo/exports.dm

```diff
@@ -48,7 +50,21 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 				else
 					E.sell_object(thing, contraband, hacked)
 					sold_str += " [thing.name]"
+				found_export = TRUE
 				break
+		if(!found_export)
+			var/item_value = thing.get_item_cost(TRUE)
+			if(item_value)
+				cost += item_value
+				sold_str += " [thing.name]"
+				if(!dry_run)
+					var/datum/export/new_export = new()
+					new_export.unit_name = thing.name
+					new_export.cost = item_value
+					new_export.include_subtypes = FALSE
+					new_export.export_types = list(thing.type)
+					new_export.sell_object(thing, contraband, hacked)
+					SSsupply.exports.Add(new_export)
```

Comment:
Why even bother generating export datums for that? Move "sell_object" elsewhere and use it without invoking export datums.

Reply:
It's to keep track of the number of exported items of the same type when making the export log in the supply console. Is there a better way to do this?